prusti-viper/src/encoder/procedure_encoder.rs

```diff
@@ -2444,12 +2511,204 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
+                    let slice_types_base = self.encoder.encode_slice_types(base_seq_ty.peel_refs())?;
+                    slice_types_base.encode_slice_len_call(self.encoder, base_seq_expr)
+                } else { todo!("Get last idx for {}", base_seq_ty) }
+            }
+            _ => unreachable!("{}", idx_ident)
+        };
+
+        trace!("start: {}, end: {}", start, end);
+
+
+
+
+        let slice_types_lhs = self.encoder.encode_slice_types(lhs_slice_ty)?;
+        let tymap = HashMap::new();
+        let elem_snap_ty = self.encoder.encode_snapshot_type(slice_types_lhs.elem_ty_rs, &tymap)?;
+
+        // length
+        let length = vir_expr!{ [end] - [start] };
+        let slice_len_call = slice_types_lhs.encode_slice_len_call(self.encoder, lhs_slice_expr.clone());
+        stmts.push(vir_stmt!{
+            inhale [vir_expr!{ [slice_len_call] == [length] }]
+        });
+
+        // lookup_pure: contents
+        let i = vir_local!{ i: Int };
+        let i_var: vir::Expr = i.clone().into();
+        let j_var: vir::Expr = j.clone().into();
+
+        let lhs_lookup_i = {
+            slice_types_lhs.encode_lookup_pure_call(
+                self.encoder,
+                lhs_slice_expr,
+                vir::Expr::from(i),
+                elem_snap_ty,
+            )
+        };
+
+        // NOTE: the lhs_ and rhs_ here refer to the moment the slice is created, so lhs_lookup is
+        // lookup on the slice currently being created, and rhs_lookup is on the array or slice
+        // being sliced
+        let lookup_eq = vir_expr!{ [lhs_lookup_i] == [rhs_lookup_j] };
+        let indices = vir_expr!{
+            [vir_expr!{ [vir::Expr::from(0)] <= [i_var] }] &&
+            (
+                [vir_expr!{ [i_var] < [slice_len_call] }] &&
+                (
+                    [vir_expr!{ [j_var] == [vir_expr!{ [i_var] + [start] }] }] &&
+                    (
+                        [vir_expr!{ [start] <= [j_var] }] &&
+                        [vir_expr!{ [j_var] < [end] }]
+                    )
+                )
+            )
+        };
+
+        // TODO: maybe don't bother with a complicated forall if the array length is less than some
+        // reasonable bound
+        stmts.push(vir_stmt!{
+            inhale [vir_expr!{ forall i: Int, j: Int :: {} ([indices] ==> [lookup_eq]) }]
+        });
+
+        self.encode_transfer_args_permissions(location, args,  &mut stmts, label.clone(), false)?;
+        // Store a label for permissions got back from the call
+        debug!(
+            "Pure function call location {:?} has label {}",
+            location, label
+        );
+        self.label_after_location.insert(location, label);
+
+        // plan
+        //
+        // [x] inhale Array$lookup_pure == Slice$lookup_pure with quantifier from start to end
+        // [?] label, encode_transfer_permissions?
+        // [ ] look at the other range types (at least those from the stdlib?
```

Comment:
```suggestion
        // [x] look at the other range types (at least those from the stdlib?
```

Reply:
Actually, if there is a "checklist" left in the comments, that should be removed.